modules/core/src/main/java/org/apache/ignite/internal/commandline/dr/subcommands/DrCacheCommand.java

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.commandline.dr.subcommands;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import org.apache.ignite.internal.commandline.CommandArgIterator;
+import org.apache.ignite.internal.commandline.dr.DrSubCommandsList;
+import org.apache.ignite.internal.util.typedef.T2;
+import org.apache.ignite.internal.visor.dr.VisorDrCacheTaskArgs;
+import org.apache.ignite.internal.visor.dr.VisorDrCacheTaskResult;
+
+import static org.apache.ignite.internal.commandline.CommandHandler.DELIM;
+import static org.apache.ignite.internal.commandline.CommandLogger.INDENT;
+
+/** */
+public class DrCacheCommand extends
+    DrAbstractSubCommand<VisorDrCacheTaskArgs, VisorDrCacheTaskResult, DrCacheCommand.DrCacheArguments>
+{
+    /** {@inheritDoc} */
+    @Override protected String visorTaskName() {
+        return "org.gridgain.grid.internal.visor.dr.console.VisorDrCacheTask";
+    }
+
+    /** {@inheritDoc} */
+    @Override public DrCacheArguments parseArguments0(CommandArgIterator argIter) {
+        String regex = argIter.nextArg("Cache name regex expected.");
+
+        try {
+            Pattern.compile(regex);
+        }
+        catch (Exception e) {
+            throw new IllegalArgumentException("Cache name regex is not valid.", e);
+        }
+
+        boolean config = false;
+        boolean metrics = false;
+        CacheFilter cacheFilter = CacheFilter.ALL;
+        SenderGroup senderGroup = SenderGroup.ALL;
+        String senderGroupName = null;
+        Action action = null;
+
+        String nextArg;
+
+        //noinspection LabeledStatement
+        args_loop: while ((nextArg = argIter.peekNextArg()) != null) {
+            switch (nextArg.toLowerCase(Locale.ENGLISH)) {
+                case "--config":
```

Comment:
May be it make sense to declare static constants for all strings like "--config" in order to have them in one place.

Reply:
fixed